### PR TITLE
Update README & remove old docker versions

### LINKED ISSUE #144: official docker image for 2021.01
Hi,

It looks everything is ready for `2021.01` image but it is not available on [docker hub](https://hub.docker.com/_/friendica?tab=description&page=1&ordering=last_updated)

If I've well understood it requires to PR something [here](https://github.com/docker-library/official-images/pulls?q=label%3Alibrary%2Ffriendica) ?

Is there another place to get image while it is approved on docker hub? :) 

Thank you

### PR DESCRIPTION
**Pull request by nupplaphil**
Fixes #144 

@MrPetovan please have a look at the README if understandable :)

### CODE REVIEW

File: README-MAINTENANCE.md [hunk 1]
```diff
@@ -6,30 +6,90 @@ The structure and usage of this repository is influenced by other, official dock
 
 **This is important!**
 
-Never ever change a file/folder inside a directory without a `.` at the beginning (`2018.05-rc`, ...).
+The `update.sh` script automatically deletes all non-`.`-directories (= version directories) at first.
+
+Never ever change a file/folder inside a directory without a `.` a t the beginning (`2018.05-rc`, ...).
```
Comment (MrPetovan) [inline README-MAINTENANCE.md]:
```suggestion
Never ever change a file/folder inside a directory without a `.` at the beginning (`2018.05-rc`, ...).
```

File: README-MAINTENANCE.md [hunk 2]
```diff
@@ -6,30 +6,90 @@ The structure and usage of this repository is influenced by other, official dock
...
+This script is a cronjob every 15 minutes and automatically runs `update.sh`.
+In case there are changes because of `update.sh`, it starts `images.yml`.
+
+### [`images.yml`](https://github.com/friendica/docker/blob/stable/.github/workflows/images.yml)
+
+This script automatically creates autotests actions based on the docker image version structure and variants.
+For example if there are two versions (like `2021.03-dev` and `2021.01`), it would create 3 runs (apache, apache-fpm, fpm-alpine) for each version, having 6 runs at all.
 
-# `update.sh`
+Therefore, the script uses the official docker github action tool [`bashbrew`](https://github.com/docker-library/bashbrew.git) to transform the repository version structure into github action commands
```
Comment (MrPetovan) [inline README-MAINTENANCE.md]:
```suggestion
The script uses the official docker GitHub Action tool [`bashbrew`](https://github.com/docker-library/bashbrew.git) to transform the repository version structure into GitHub Action commands.
```

File: README-MAINTENANCE.md [hunk 3]
```diff
@@ -6,30 +6,90 @@ The structure and usage of this repository is influenced by other, official dock
...
+For mor details see [Introduction to github actions](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/introduction-to-github-actions).
+
+### [`update-sh.yml`](https://github.com/friendica/docker/blob/stable/.github/workflows/update-sh.yml)
+
+This script is a cronjob every 15 minutes and automatically runs `update.sh`.
+In case there are changes because of `update.sh`, it starts `images.yml`.
+
+### [`images.yml`](https://github.com/friendica/docker/blob/stable/.github/workflows/images.yml)
+
+This script automatically creates autotests actions based on the docker image version structure and variants.
```
Comment (MrPetovan) [inline README-MAINTENANCE.md]:
```suggestion
This script automatically creates autotests actions based on the Docker image version structure and variants.
```

File: README-MAINTENANCE.md [hunk 4]
```diff
@@ -6,30 +6,90 @@ The structure and usage of this repository is influenced by other, official dock
...
+
+1. At first, read https://github.com/docker-library/docs#how-do-i-update-an-images-docs
+2. Checkout & change the content of https://github.com/docker-library/docs/tree/master/friendica
+3. Create a new Pull Request at https://github.com/docker-library/docs/pulls 
+
+# Tools & Scripts
+
+## Github Action
+
+GitHub Actions helps to automate tasks for Continuous Integration (= autotest docker images) and Continuous Deployment (= autocreate docker images based on the friendica upstream).
```
Comment (MrPetovan) [inline README-MAINTENANCE.md]:
```suggestion
GitHub Actions helps to automate tasks for Continuous Integration (= autotest Docker images) and Continuous Deployment (= autocreate Docker images based on the friendica upstream).
```

File: README-MAINTENANCE.md [hunk 5]
```diff
@@ -6,30 +6,90 @@ The structure and usage of this repository is influenced by other, official dock
...
+
+Never ever change a file/folder inside a directory without a `.` a t the beginning (`2018.05-rc`, ...).
 This folder will get updated automatically based on the templates you want to change.
 All changes in such folders will get overwritten during an update.
 
-# Maintenance scripts
+# How to update
+
+This section describes how to update the official Friendica docker images.
+The official docker image is available at https://hub.docker.com/_/friendica.
```
Comment (MrPetovan) [inline README-MAINTENANCE.md]:
```suggestion
The official Docker image is available at https://hub.docker.com/_/friendica.
```

File: README-MAINTENANCE.md [hunk 6]
```diff
@@ -6,30 +6,90 @@ The structure and usage of this repository is influenced by other, official dock
...
-# `generate-stackbrew-library.sh`
-   
-This file automatically creates a "manifest" for the docker-images.
+Any other pattern will completely be ignored(!)
+
+#### Minimum version
+
+Please adjust the `min_version` variable.
+It contains the minimum supported version, which will automatically be available as a docker image.
+Any other versions lower than this won't be available from the official docker hub anymore.
```
Comment (MrPetovan) [inline README-MAINTENANCE.md]:
```suggestion
It contains the minimum supported version, which will automatically be available as a Docker image.
Any other versions lower than this won't be available from the official Docker Hub anymore.
```

File: README-MAINTENANCE.md [hunk 7]
```diff
@@ -6,30 +6,90 @@ The structure and usage of this repository is influenced by other, official dock
...
+
+### [`generate-stackbrew-library.sh`](https://github.com/friendica/docker/blob/stable/generate-stackbrew-library.sh)
+
+This file creates a "manifest" for the docker-images.
 This "manifest" is used to create a new PR in the official-images [repository](https://github.com/docker-library/official-images/) for deploying the changes to the docker hub.
```
Comment (MrPetovan) [inline README-MAINTENANCE.md]:
```suggestion
This "manifest" is used to create a new PR in the official-images [repository](https://github.com/docker-library/official-images/) for deploying the changes to the Docker Hub.
```

File: README-MAINTENANCE.md [hunk 8]
```diff
@@ -54,3 +114,9 @@ GitCommit: b46fae917321394e1482df59dc4e39daffbe5c59
 Directory: 2018.05-rc/fpm-alpine
 This is the input-file for the official-images in a later step :-)
 ```
+
+#### Release channels
+
+Please adjust the `release_channel` array.
+It maps additional tags onto docker image versions.
```
Comment (MrPetovan) [inline README-MAINTENANCE.md]:
```suggestion
It maps additional tags onto Docker image versions.
```

File: generate-stackbrew-library.sh [hunk 9]
```diff
@@ -2,7 +2,7 @@
 set -Eeuo pipefail
 
 declare -A release_channel=(
-  [stable]='2020.09'
+  [stable]='2021.01'
```
Comment (MrPetovan) [inline generate-stackbrew-library.sh]: Can't we have this in a `stable.txt` file like the `latest` and `develop` in the same file? Or can't we just use the `latest`?

**Review by MrPetovan — CHANGES_REQUESTED** [review]

**Review by MrPetovan — APPROVED** [review]

**Comment by MrPetovan** [comment]
The GitHub Action Stackbrew manifest step failed: https://github.com/friendica/docker/pull/145/checks?check_run_id=1652102214

**Comment by nupplaphil** [comment]
I removed it again .. I'll better create a additional PR for it ;-)
=> will remove the need of a local docker/bashbrew environment

**Comment by valvin1** [comment]
thank you :) my instance is now with 2021.01 version

**Pull request merged**